### PR TITLE
refactor(sdk)!: migrate evaluate() onto the result envelope

### PR DESCRIPTION
**Pull request by adnanrhussain**
Closes the largest conformance gap: §5.1's result envelope.

## The shape

```ts
// before
{ score, reasoning, metadata: { model, processingTimeMs, inputTokens, outputTokens }, _internal }

// after
{ evaluator, result, metadata: { model, processingTimeMs, tokenUsage: { inputTokens, outputTokens } } }
```

`result` is the model's structured output as the evaluator's `output_schema.json` declares it — keys and values unaltered, no scalar hoisted up, no casing translation.

That last point is deliberate and worth a reviewer's attention. §5.2 says field names follow §2, which maps snake_case to camelCase for the SDK surface. I read that as **not** applying to the payload, because §5.2 also requires the payload be *identical across SDKs* — and casing-mapping it would give Python `complexity_score` and TypeScript `complexityScore`, i.e. two different payloads for the same evaluator. Keeping the registry's own keys is the only reading that satisfies both sentences. Worth confirming in the spec.

## The display maps are gone

Callers now see `slightly_complex`, not `Slightly complex`. Three per-evaluator lookup tables deleted — exactly the kind of thing a second SDK would have to clone byte-for-byte or drift on.

## readOutcome

Reports need one comparable value per evaluation and the envelope has no score slot, so this derives it from the shape the contracts already share: the single required `*_score` property plus `reasoning`. A new evaluator that follows the convention needs no registration — the feedback family's `quality_score` works untouched, and there's a test asserting that.

Two declared exceptions, both with a reason in the code:

| evaluator | field | why |
| --- | --- | --- |
| Grade Level Appropriateness | `grade` | answers with a band, not a score |
| Sentence Structure | `answer` | until its schema is generated from its contract |

`Outcome.score` is `string | undefined`: a payload with no verdict field reports the gap rather than rendering it as `''`, since a silent fallback is what Principle 4 calls a defect. Batch still renders blank, but coerces at its own call site where the gap stays visible. `readOutcome` takes the envelope — it already carries `evaluator`, so making the caller disassemble it was needless.

## One place data is dropped

**Sentence Structure's payload narrows.** Its `output_schema.json` declares `complexity_score` and `reasoning` only, and calls the analysis step's ~30 grammatical counts *"an internal intermediate artifact, not part of this contract"* — so they are no longer returned. This was a real choice between the contract and §5.2's "never withhold data the model returned"; the contract won because §5.2 also says the payload's shape *is* the registry definition. Flagging it as the one caller-visible capability loss.

## Out of scope

- **Math Standards Alignment** shares none of these types — it has its own `StandardAlignmentResult` and never used `EvaluationResult`. It needs the envelope too, in its own PR.
- **`evaluateItems` / `evaluateByGradeLevel`** stay as they are. §9.1 is Draft and Q-12 explicitly lists batch result shapes as undecided, so settling them here would be inventing a contract.
- **Per-step detail in `metadata`.** A multi-step evaluator's envelope shows only sums: `tokenUsage` across all steps, wall-clock `processingTimeMs`, one `model` string. Every evaluator already builds a per-step `stageDetails` array, but it goes to telemetry only — so the breakdown exists and the caller can't see it. The spec-legal home is `metadata`, but §8.3 phase details are Draft pending Q-10, which covers the `stage`/`step`/`phase` naming itself. Adding it now would commit to that vocabulary, inside the one structure §5.1 says is fixed forever.

## Breaking

Every `evaluate()` return shape; `_internal` → `result`; flat metadata token counts → `metadata.tokenUsage`; score values for Purpose Clarity, Organizational Structure and Reference Knowledge Demands; and three removed exports — `SentenceStructureInternal`, `PurposeClarityComplexityLevel`, and the second type parameter of `EvaluationResult`.

## Validation

- `tsc --noEmit`, `npm run lint`, `npm run test:unit`, `npm run build`
- `scripts/check.py` from the repo root
- StrykerJS on `src/schemas/outcome.ts` — one surviving mutant, equivalent (with `field` undefined, `result[field]` is `undefined` on both branches, so Stryker flags it whichever way the ternary is written)
- The live cross-provider integration test now asserts through `readOutcome`, so it exercises every evaluator's payload shape in one place

### CODE REVIEW

**Comment by adnanrhussain** [comment]
All three comments were correct. Fixed in fc2bb665.

**1. Batch aggregates silently dropped the new score values** — the important one. `COMPLEXITY_SCORE_MAP` is keyed on the spaced form and looked up with `score.toLowerCase().trim()`, so `slightly_complex` missed and those rows vanished from every complexity average and from the heatmap, with no error. I reproduced it first: a two-row batch scoring `slightly_complex` and `very_complex` gave an average of **0** instead of 2.

Fixed by folding underscores at the lookup rather than normalizing in the family, so the SDK's returned value stays the contract's and only the report's own numeric mapping accepts both spellings. Two tests: one for the contract form, one for the spaced form that four evaluators still return until their schemas are generated.

**2. `readOutcome` preferred an absent override.** Real forward-compat trap — when Sentence Structure's schema is regenerated to `complexity_score`, the stale `answer` entry would have taken precedence and the verdict would have gone silently empty. An override now wins only while its field is present, so the entry becomes inert instead of harmful. Added a test for exactly that transition.

Also tightened `isRecord` to exclude arrays, with a case in the table-driven test. No behaviour change (an array payload already yielded blanks) but the guard now says what it means.

**3. Duplicated assertions** — correct, lines 142–143 already asserted the same pair. Replaced mine with the assertion that actually adds coverage: that the analysis step's counts are *not* in the payload, which is the narrowing this PR introduces.

Mutation testing on `outcome.ts` after the fixes: the precedence change initially left two `ConditionalExpression` survivors, both because an `undefined` guard is redundant next to `in` (which coerces its operand, so `undefined in result` is just `'undefined' in result`). Rather than add a contrived test I restructured around an empty-string sentinel, which removed both branches. Both conditional survivors are now killed; the two that remain are the sentinel's value being arbitrary, which is equivalent by construction.

File: sdks/typescript/src/schemas/outcome.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * than throwing — a missing verdict is a reporting gap, not a reason to fail an
+ * evaluation that already succeeded.
+ */
+export function readOutcome(evaluatorId: string, result: unknown): Outcome {
+  if (!isRecord(result)) return { score: '', reasoning: '' };
+
+  const override = VERDICT_FIELD_OVERRIDES[evaluatorId];
+  const field = override ?? Object.keys(result).find((key) => key.endsWith('_score'));
+
+  const score = field !== undefined ? result[field] : undefined;
```
[the remark this answers is not in the document]
Comment (adnanrhussain) [reply sdks/typescript/src/schemas/outcome.ts]: **Addressed** in fc2bb665. An override now wins only while its field is present, so a stale entry becomes inert rather than blanking the verdict. `isRecord` also excludes arrays.

File: sdks/typescript/tests/unit/evaluators/sentence-structure.test.ts [hunk 2]
```diff
@@ -139,23 +139,22 @@ describe('SentenceStructureEvaluator - Evaluation Flow', () => {
...
 
-      // Verify internal data structure
-      expect(result._internal).toBeDefined();
-      expect(result._internal?.sentenceAnalysis).toBeDefined();
-      expect(result._internal?.features).toBeDefined();
-      expect(result._internal?.complexity).toBeDefined();
+      // The payload is the classify step's output, which is all output_schema.json
+      // declares -- the analysis step's counts are an internal intermediate.
+      expect(result.result.answer).toBe('Slightly complex');
+      expect(result.result.reasoning).toContain('simple sentence structures');
```
[the remark this answers is not in the document]
Comment (adnanrhussain) [reply sdks/typescript/tests/unit/evaluators/sentence-structure.test.ts]: **Addressed** in fc2bb665. Replaced the duplicated pair with an assertion that the analysis counts are absent from the payload.

File: sdks/typescript/src/batch/families/qtc.ts [hunk 3]
```diff
@@ -98,7 +99,7 @@ class QtcRunner implements FamilyRunner {
 
   async runTask(row: FamilyRow, memberId: string): Promise<TaskOutcome> {
     const result = await this.getEvaluator(memberId).evaluate(row.columns.text, row.columns['grade_level']);
-    return { score: result.score, reasoning: result.reasoning };
+    return readOutcome(result.evaluator, result.result);
   }
```
[the remark this answers is not in the document]
Comment (adnanrhussain) [reply sdks/typescript/src/batch/families/qtc.ts:107]: **Addressed** in fc2bb665, though in `formatters.ts` rather than here: folding underscores at the score lookup keeps the SDK returning the contract value instead of reintroducing a display transform at the family boundary. Reproduced first — the aggregate was 0 instead of 2.
